pkg/deployer/lib/readinesscheck/readiness.go

```diff
@@ -108,18 +113,25 @@ func (e *ObjectNotReadyError) Error() string {
 }
 
 // IsObjectReady gets an updated version of an object and checks if it is ready.
-func IsObjectReady(ctx context.Context, kubeClient client.Client, obj *unstructured.Unstructured, checkObject checkObjectFunc) error {
+func IsObjectReady(ctx context.Context, kubeClient client.Client, obj *unstructured.Unstructured,
+	checkObject checkObjectFunc, isCheckRelevant isCheckRelevantFunc, failOnMissingObject bool) error {
 	objLog, ctx := logging.FromContextOrNew(ctx, nil,
 		lc.KeyGroupVersionKind, obj.GroupVersionKind().String(),
 		lc.KeyResource, kutil.ObjectKey(obj.GetName(), obj.GetNamespace()).String())
 
+	if !isCheckRelevant(obj) && !failOnMissingObject {
+		return nil
+	}
+
 	key := kutil.ObjectKey(obj.GetName(), obj.GetNamespace())
 	if err := kubeClient.Get(ctx, key, obj); err != nil {
-		objLog.Debug("Resource status", lc.KeyStatus, StatusUnknown)
-		return fmt.Errorf("unable to get %s %s/%s: %w",
-			obj.GroupVersionKind().String(),
-			obj.GetName(), obj.GetNamespace(),
-			err)
+		if !errors.IsNotFound(err) || isCheckRelevant(obj) || failOnMissingObject {
```

Comment:
as we have the check in line 122 `if !isCheckRelevant(obj) && !failOnMissingObject {` we could remove the complete check here in line 128 `if !errors.IsNotFound(err) || isCheckRelevant(obj) || failOnMissingObject {`

Reply:
Correct.

This means we don't need to pass the `isCheckRelevant` function and the `failOnMissingObject` flag through all the methods. We could simply filter the object list earlier, and call `WaitForObjectReady` for the already filtered object list.